ldj-spring-renderer-example/src/main/java/com/infosupport/ldoc/springexample/ampq/SpringAmpqRenderer.java

```diff
@@ -0,0 +1,234 @@
+   *
+   * @param node      Node to check
+   * @param className class name that is attached to interactions that are found
+   * @return A list of Post Queue interactions that where found
+   */
+  public static List<QueueInteraction> findQueuePosts(JsonNode node, String className) {
+    List<QueueInteraction> interactions = new ArrayList<>(0);
+
+    QueueInteraction found = extractQueuePost(node, className);
+    if (found != null) {
+      interactions.add(found);
+    }
+
+    for (JsonNode child : node) {
+      interactions.addAll(findQueuePosts(child, className));
+    }
+    return interactions;
+  }
+
+  /**
+   * Presumes node is array type.
+   *
+   * @param nodes JsonNode (presumed to be of Json Array type
+   * @param key   the requested key of the JsonNode
+   * @param value the requested value of the JsonNode
+   * @return Json node (object) which contains value associated to key
+   */
+  public static JsonNode getNodeWithKeyAndValue(JsonNode nodes, String key, String value) {
+    if (!nodes.isArray()) {
+      return null;
+    }
+
+    for (JsonNode node : nodes) {
+      if (node.path(key).asText().equals(value)) {
+        return node;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * From a JsonNode, gather all QueueInteractions.
+   *
+   * @param root root Node of the Json tree corresponding to a LivingDocumentation analysis Json
+   *             file
+   * @return a list of all top-level interactions (note that post interactions that happen in a read
+   *             interaction are nested)
+   */
+  public static List<QueueInteraction> findAllQueueInteractions(JsonNode root) {
+    List<QueueInteraction> interactions = new ArrayList<>();
+
+    for (JsonNode typeDesc : root) {
+      String className = typeDesc.path("FullName").textValue();
+
+      //check if class is listener
+      JsonNode rabbitListenerAnnotation = containsAttribute(typeDesc, RABBIT_LISTENER_ANNOTATION);
+      if (rabbitListenerAnnotation != null) { // I.e., this class is a RabbitListener
+        // Note that the queue is the same for all methods within a RabbitListener
+        JsonNode queueArgument = getNodeWithKeyAndValue(rabbitListenerAnnotation.path("Arguments"),
+            "Name", "queues");
+        assert queueArgument
+            != null; // TODO This should not happen, how nicely do we want to handle this?
+        String queue = queueArgument.path("Value").asText();
+
+        //Check for listen methods
+        for (JsonNode method : typeDesc.path("Methods")) {
+          if (isQueueReadMethod(method)) {
+            // Note the type is stripped to support messages across different applications
+            String messageType = stripName(
+                method.path("Parameters").path(0).path("Type").textValue());
+            // Create and add ReadInteraction
+            QueueInteraction newRead = new QueueInteraction(className, QueueInteractionKind.READ,
+                queue, messageType, new ArrayList<>(0));
```

Comment:
Agreed based on code convention, but I think that would make the code less understandable for someone reading this renderer for a first time. In case the model changes and interactions become classes this would not be a problem right?